### PR TITLE
Make version informations consistent between prometheus components

### PR DESCRIPTION
**Pull request by sdurrheimer**
~~There is one breaking change, the `/version` API route now return CamelCase field names instead of lowerCamelCase.~~

### CODE REVIEW

File: storage/local/storagetool/main.go [hunk 1]
```diff
@@ -14,14 +14,12 @@
 package main
 
 import (
-	"bytes"
 	"fmt"
 	"os"
-	"strings"
-	"text/template"
+
+	"github.com/prometheus/common/version"
```
Comment (grobie) [inline storage/local/storagetool/main.go]: What's with all the newlines here?

Comment (sdurrheimer) [inline storage/local/storagetool/main.go]: Some existing imports were like that. I will fix this.

File: web/web.go [hunk 2]
```diff
@@ -110,6 +111,40 @@ func (s *PrometheusStatus) ApplyConfig(conf *config.Config) bool {
...
+	Version   string
+	Revision  string
+	Branch    string
+	BuildUser string
+	BuildDate string
+	GoVersion string
+}
+
+// Map provides the iterable version information.
+func (pv *PrometheusVersion) Map() map[string]interface{} {
```
Comment (grobie) [inline web/web.go]: This looks overkill to me. Let's just change the template to use the keys explicitly and remove the iteration.

File: web/web.go [hunk 3]
```diff
@@ -110,6 +111,40 @@ func (s *PrometheusStatus) ApplyConfig(conf *config.Config) bool {
 	return true
 }
 
+// PrometheusVersion contains various build information about Prometheus
+type PrometheusVersion struct {
+	Version   string
```
Comment (grobie) [inline web/web.go]: Let's be explicit about the json names `Version string `json:"version"`. The current version will change the keys from lowercase to uppercase. See http://demo.robustperception.io:9090/version

Comment (sdurrheimer) [inline web/web.go]: That's what I've said in the description, but thanks I didn't though about specifying json names in the struct. I will fix this.

File: web/web.go (hunk 2 above)
Comment (sdurrheimer) [inline web/web.go]: Yeah, that's what I thought too. I have put it anyway to see what you guys think about this. I will do the change.

**Comment by sdurrheimer** [comment]
Updated

**Comment by grobie** [comment]
LGTM :+1: @juliusv

File: .promu.yml [hunk 4]
```diff
@@ -9,11 +9,11 @@ build:
           path: ./cmd/promtool
     flags: -tags netgo
     ldflags: |
-        -X {{repoPath}}/version.Version={{.Version}}
-        -X {{repoPath}}/version.Revision={{.Revision}}
-        -X {{repoPath}}/version.Branch={{.Branch}}
-        -X {{repoPath}}/version.BuildUser={{user}}@{{host}}
-        -X {{repoPath}}/version.BuildDate={{date "20060102-15:04:05"}}
+        -X {{repoPath}}/vendor/github.com/prometheus/common/version.Version={{.Version}}
```
Comment (fabxc) [inline .promu.yml]: Really not a fan of this. But if we want it in a library there's probably not better option.

Comment (grobie) [inline .promu.yml]: We could keep the variables in each repository and use a struct like `Info` to pass it to the print, collector, etc. functions.

Comment (sdurrheimer) [inline .promu.yml]: Which will add duplicated lines of code in each repository.
Indeed those flags looks nasty, but there no other choise for a vendored lib.
Moreover, this is a one time definition in the `.promu.yml` file, looks ugly but it works.

Comment (fabxc) [inline .promu.yml]: Yea, if we do the former we could just duplicate everything.

**Comment by fabxc** [comment]
👍

**Pull request merged**